{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "380321ea-f2e1-4c50-afa0-6300e65bea15",
   "metadata": {},
   "source": [
    "# Project 3 Report\n",
    "**Xiaohe Jin Nov-23-2025** "
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "b7083059-ab20-492d-be9a-97f54c35f841",
   "metadata": {},
   "source": [
    "## Introduction:\n",
    "RUNX1 is a pivotal member of the RUNX family of transcription factors, originally characterized for its essential role in hematopoiesis and leukemia development(1). Recent studies indicate that RUNX1 also plays significant and complex roles in breast cancer, where its altered expression and recurrent mutations have been observed, though its precise functions remain uncovered. This study(2) addresses the critical question of how RUNX1 contributes to breast cancer biology by exploring the connection between RUNX1-dependent gene regulation and chromatin organization in MCF-7 breast cancer cells. To answer this biological question, the authors applied an integrative suite of high-throughput bioinformatic techniques: ChIP-seq to define genome-wide RUNX1 binding sites, RNA-seq to profile transcriptional changes following RUNX1 depletion, and Hi-C to probe whether RUNX1 modulates topologically associating domains (TADs) or influences long-range chromatin contacts. Overall, the study provides new insight into RUNX1's regulatory networks and its architectural role in local chromatin structure, expanding our understanding of how disruptions in RUNX1 may drive breast cancer progression and impact genome regulation."
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "4e2f4a5e-7ea0-4405-b833-279f21aa388c",
   "metadata": {},
   "source": [
    "## Methods:\n",
    "Raw ChIP-seq data (GEO accession: GSE75070) for RUNX1 in parental MCF-7 cells were processed using a modular workflow implemented in Nextflow (v25.04.6)(3). All bioinformatics software tools were executed within Docker containers(4) specified for each process, ensuring computational reproducibility, portability, and consistency across different computing environments.\n",
    "    \n",
    "### Adapter Trimming, Genome Mapping, and Quality Control\n",
    "Initial sequencing read quality was assessed using FastQC (v0.12.1) ((https://github.com/s-andrews/FastQC). Adapters and low-quality bases were then trimmed from the reads using Trimmomatic(5) (v0.39) with the parameters: *-phred33 ILLUMINACLIP:${adapter.fa}:2:30:10:2:True LEADING:3 TRAILING:3*. The adapter sequences used for trimming were: (1) TruSeq3 Indexed Adapter (AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC) and (2) TruSeq3 Universal Adapter (AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAT). The trimmed reads were aligned to the human reference genome (GRCh38/hg38) using Bowtie2 (v2.5.4)(6). The resulting Sequence Alignment Map (SAM) files were converted to Binary Alignment Map (BAM) format, sorted, and indexed using SAMtools (v1.22.1)(7) with default settings. Alignment statistics, including mapping rates, were generated using SAMtools flagstat. Quality control reports from FastQC, Trimmomatic, and SAMtools flagstat were aggregated into a single interactive report using MultiQC (v1.25)(8). For downstream visualization and inter-sample correlation analysis, normalized coverage tracks in BigWig format were generated using deepTools bamCoverage (v3.5.5)(9). This process calculated the number of sequencing reads per genomic bin for each sample, creating files suitable for profiling read density across specific genomic loci.\n",
    "\n",
    "### Sample Correlation Analysis\n",
    "A correlation analysis was performed to assess the similarity between replicate samples based on their genome-wide coverage profiles. The multiBigwigSummary utility from deepTools (v3.5.5)(9) was used to compute the read coverage values across genomic bins for all BigWig files, generating a summary matrix. This matrix was then used as input for the plotCorrelation utility, which was executed with the *-c pearson* parameter to calculate Pearson correlation coefficients and generate a heatmap visualizing the pairwise relationships between samples. The Pearson method was selected for its sensitivity in measuring linear relationships between coverage scores, providing a robust assessment of replicate concordance.\n",
    "\n",
    "### RUNX1 Peak Calling and Replicate Concordance\n",
    "Peak calling was conducted separately for each biological replicate using HOMER (v4.11)(10). First, makeTagDirectory utility was used to create tag directories from the aligned BAM files. Subsequently, RUNX1 binding peaks were identified for each replicate using the findPeaks script with the -style factor parameter and the following thresholds: *-F 1.5 -P 0.001 -L 1.5 -LP 0.001 -minDist 200 -size 300 -inputSize 600*. This command was executed for each paired set of immunoprecipitation (IP) and corresponding control (Input) samples (e.g., IP_rep1 vs. Input_rep1). The resulting peak locations from the HOMER text output files were converted to the standard BED format using the pos2bed.pl script.\n",
    "\n",
    "To generate a single, high-confidence set of reproducible RUNX1 peaks, the BED files from the two biological replicates were intersected using BEDTools intersect (v2.31.1)(11) with a permissive parameter of *-f 0.0001*, requiring only a single base pair overlap to define a peak as reproducible. This inclusive approach aimed to maximize sensitivity in capturing all potential binding sites. The resulting peak set was then filtered to remove any peaks residing in artifactual or high-signal regions using the ENCODE hg38 consensus blacklist (v2) with BEDTools intersect and the -A parameter, which removes any peak that overlaps a blacklisted region by even one base pair.\n",
    "\n",
    "### Genomic Annotation of RUNX1 Peaks\n",
    "The final set of reproducible, blacklist-filtered peaks was annotated to determine their genomic context and proximity to known features. This was performed using the annotatePeaks.pl script from HOMER (v4.11)(10), which was provided with the peak BED file, the reference genome sequence (GRCh38), and its corresponding gene annotation file (gtf). This analysis assigned each peak to its nearest transcriptional start site or genomic feature, providing initial functional insight into the potential regulatory targets of RUNX1.\n",
    "\n",
    "### Signal Profiling Across Gene Bodies\n",
    "To visualize the aggregate RUNX1 binding signal relative to gene architecture, a meta-profile was generated using deepTools (v3.5.5) with scale-regions utility. The window size was set as 2,000 bp to calculate coverage values across the body of all annotated genes. The analysis was performed exclusively on the immunoprecipitation (IP) samples using a BED file of hg38 gene annotations obtained from the UCSC Genome Browser. The resulting matrix was then passed to the plotProfile utility to generate a composite plot depicting the average RUNX1 ChIP-seq signal intensity across the genic regions and their immediate flanking sequences.\n",
    "\n",
    "### De Novo Motif Discovery\n",
    "*De novo* motif enrichment analysis was performed on the final set of reproducible, blacklist-filtered RUNX1 peaks to identify significantly overrepresented DNA sequence motifs. This analysis was carried out using the findMotifsGenome.pl script from HOMER (v4.11), with the peak BED file and the GRCh38 reference genome as inputs. The region size was set to 200 bp, and the resulting motif annotations and statistical outputs were saved for downstream interpretation."
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "b3bee760-fefb-4a55-b002-605d5783eddd",
   "metadata": {},
   "source": [
    "## Results\n",
    "### Quality Control Evaluation\n",
    "Initial quality assessment of the two RUNX1 ChIP (IP_rep1 and IP_rep2) libraries and their matched inputs showed overall high read quality, with most bases exceeding Phred 30. After adapter and low-quality trimming with Trimmomatic, the surviving read counts ranged from 10.9 million to 30.0 million per sample, with minimal loss in the IP libraries (2.3% and 3.4%). FASTQC reported only minor overrepresented sequences, ranging from 2.1% to 3.1%, which is primarily adapter-derived and were effectively removed after trimming. No additional quality warnings indicated persistent sequence bias or contamination.\n",
    "\n",
    "Alignment of the trimmed reads to the hg38 reference genome was highly efficient, with mapping rates ranging from 92.8% (INPUT_rep2) to 98.1% (IP_rep2). As expected for a successful ChIP-seq experiment, the IP samples showed a markedly higher duplication rate (74.3% and 89.1%) compared to the input controls (10.6% and 13.0%), reflecting the enrichment of a specific subset of genomic fragments. Furthermore, the IP samples exhibited a higher GC content (47%) than the inputs (43%), which is consistent with the binding of a transcription factor to GC-rich promoter regions. Together, the strong base quality, low contamination, high mapping efficiency, and appropriate enrichment signatures indicate that the sequencing data are of high quality and suitable for downstream analyses. The only potential improvement for future experiments would be to reduce duplication levels by increasing library complexity or reducing PCR amplification cycles.\n",
    "\n",
    "### Signal Coverage \n",
    "To examine the genomic distribution of RUNX1 binding, meta-gene profiles were generated for each immunoprecipitated (IP) sample by averaging ChIP-seq signal across all annotated gene bodies and their flanking regions **(Figure 1)**. Both replicates showed a sharp and narrow peak of RUNX1 enrichment precisely at the transcription start site (TSS), with signal intensity decreasing rapidly across the gene body and approaching background levels downstream of the transcription end site (TES). This strong promoter-proximal enrichment indicates that RUNX1 predominantly binds near TSS regions rather than distributing broadly across gene bodies or distal intergenic regions.\n",
    "\n",
    "Biologically, this pattern is characteristic of a transcription factor that acts primarily in promoter regulation, positioning RUNX1 to directly influence transcription initiation of its target genes in MCF-7 cells. This TSS-focused binding pattern supports the hypothesis that RUNX1 plays a direct and sequence-specific role in modulating gene expression programs relevant to breast cancer biology."
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "ea5e2af5-096e-4d13-8000-66f002cb0521",
   "metadata": {},
   "source": [
    "![alt_text](Figure1_Signal_Coverage.png)\n",
    "**Figure 1. Signal coverage of genomic distribution of RUNX1 binding.**"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "1ab3b2ff-73ed-4f6d-b520-c1e111c1db8c",
   "metadata": {},
   "source": [
    "### De Novo Motif Finding\n",
    "De novo motif discovery was performed on the set of high-confidence RUNX1 peaks to identify enriched DNA binding sequences **(Figure 2)**. The most significantly enriched motif was identified as the binding site for RUNX2, a member of the same RUNX transcription factor family as RUNX1. This finding is biologically plausible and can be attributed to the high degree of sequence similarity between the DNA-binding Runt domains of RUNX1 and RUNX2, which recognize nearly identical core consensus sequences. Because RUNX2 is abundantly expressed in MCF-7 breast cancer cells, the recovered motif likely reflects shared occupancy by RUNX1 and RUNX2 rather than RUNX2-specific binding.\n",
    "\n",
    "Beyond the RUNX motif, the analysis identified several other significantly enriched sequences, providing insight into potential co-regulatory networks. Notably, motifs for known oncogenic transcription factors such as ELK4 and TBP3 were highly ranked. The presence of the ELK4 motif is particularly interesting, as it is implicated in breast cancer proliferation and can co-regulate target genes with other factors. Similarly, TBP3 is a driver gene in breast cancer. Although our de novo motif analysis identified a distinct set of co-enriched factors, highlighting RUNX2, ELK4, and TBX3 rather than the STAT3 and FOXP3 reported in the original study, the core biological implication remains consistent. The use of a different algorithm (HOMER vs. MEME-ChIP) and parameters can preferentially detect motifs with varying stringencies and nucleotide compositions. Despite these technical differences, both analyses converge on the same conclusion: RUNX1 functions within a complex transcriptional landscape, potentially cooperating with other cancer-associated factors at shared genomic regulatory elements."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "64702f7a-b5bd-4ade-9c95-f75253291666",
   "metadata": {},
   "source": [
    "![alt_text](Figure2_Motif_Find.png)\n",
    "**Figure 2. De novo motif analysis of the RUNX1 peaks.**"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "ee629ab7-9309-41c3-8b64-4007cb871987",
   "metadata": {},
   "source": [
    "### Figure 2F Reproduce: Overlap the ChIPseq results with the original RNAseq data\n",
    "To functionally link RUNX1 chromatin binding with its role in gene regulation, the ChIP-seq peak set was intersected with the RNA-seq data from RUNX1-depleted MCF-7 cells published in the original study. The reproduced analysis (Figure 3) successfully recapitulated the core finding of the original publication: RUNX1 binding is markedly enriched near genes that are down-regulated following RUNX1 knockdown. This pattern is consistent across both promoter-proximal regions (±5 kb of the TSS) and broader genic regions (±20 kb of the gene body), strongly supporting the hypothesis that RUNX1 functions primarily as a transcriptional activator for these target genes.\n",
    "\n",
    "While the overall conclusion is consistent, there is a quantitative difference in my results, that a lower number of RUNX1-bound genes within the ±20 kb gene body region compared to the original Figure 2F. This discrepancy can be attributed to several technical factors inherent in reproducible computational biology. First, differences in peak-calling algorithms and stringency parameters can lead to variations in the final peak set, particularly for broader or lower-affinity binding sites often found in enhancer regions. Second, the precise definition and calculation of a peak's association with a \"gene body\" can vary between annotation tools and pipelines, affecting which peaks are assigned to this category. Despite these differences, my results potentially offer a more focused biological insight: the stronger promoter-proximal enrichment observed in our analysis is consistent with RUNX1’s well-established role as a promoter-focused transcription factor, suggesting that our pipeline may preferentially capture the highest-confidence and most direct RUNX1 regulatory interactions."
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "8e8ded36-1245-45a7-b7b2-25a137595353",
   "metadata": {},
   "source": [
    "![alt_text](Figure3.png)\n",
    "**Figure 3. Bat plot showing RUNX1 peak binding of up- and down-regulated genes.**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a5035423-571d-4662-8f97-9f171e9f9251",
   "metadata": {},
   "source": [
    "### Figure 2D and 2E Reproduce: Genomic Confirmation of RUNX1 Binding at NEAT1 and MALAT1 Promoters\n",
    "To validate specific RUNX1 targets highlighted in the original study, we examined our ChIP-seq coverage tracks at the NEAT1 and MALAT1 loci. As shown in **Figure 4–5**, the genome browser visualizations(12) confirm the presence of strong and reproducible RUNX1 binding peaks at both promoters, successfully recapitulating the key findings of the original Figures 2D and 2E. In both cases, RUNX1 enrichment is sharply localized to the transcription start site (TSS), with little to no signal across the gene bodies. This promoter-focused pattern is characteristic of direct transcriptional regulation and supports the hypothesis that RUNX1 acts at these loci to influence transcription initiation.\n",
    "\n",
    "Although the binding locations match the original figures, the signal intensity and peak width in our IGV tracks differ slightly from those presented in the publication. These differences are likely the result of variations in data normalization, scaling, or visualization settings rather than true biological discrepancies. Importantly, the precise concordance in the genomic position of RUNX1 binding reinforces the mechanistic interpretation derived from both studies: RUNX1 is positioned to directly activate NEAT1 and MALAT1 transcription. Combined with the RNA-seq evidence showing down-regulation of these lncRNAs following RUNX1 knockdown, our reproduced results further support their classification as direct RUNX1 targets in MCF-7 cells."
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "0fde9e9a-8c23-4127-a412-e9069b487c3e",
   "metadata": {},
   "source": [
    "![alt_text](Figure4_MALAT1.png)\n",
    "![alt_text](Figure5_NEAT1.png)\n",
    "**Figure 4–5: Two examples of ChIP-seq genome browser views of RUNX1 binding and the input control for MALAT1 (above) and NEAT1 (bottom) lncRNA genes.**"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "9f8a6eb1-e849-4fec-9f72-721d9e2baed7",
   "metadata": {},
   "source": [
    "### Quality Assessment and Reproducibility of RUNX1 ChIP-seq\n",
    "A comparative analysis was performed to assess the reproducibility of our ChIP-seq processing pipeline relative to the original publication, with read-count and alignment statistics summarized in Table 1. Although the total read counts closely matched those reported in the original study, our pipeline yielded substantially higher mapping rates across all samples. For example, the INPUT_rep1 library achieved a 95.7% alignment rate in our workflow compared with 63.3% in the original analysis. This improvement is likely due to two main factors: (1) the use of updated alignment tools (Bowtie2) and a modern, well-curated reference genome (GRCh38), both of which enhance read placement accuracy; and (2) differences in preprocessing stringency, including more effective adapter and low-quality base trimming, which remove reads that might otherwise map ambiguously. These refinements likely produced a cleaner and more mappable dataset, improving overall alignment performance without altering biological conclusions."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3751bd75-be19-4070-8223-79d6dc2b98c6",
   "metadata": {},
   "source": [
    "| Sample     | Biological Replicate | Total Reads | Mapped Reads | Mapping Rate |\n",
    "|------------|---------------------|------------:|-------------:|-------------:|\n",
    "| INPUT_rep1 | Rep1                | 12,345,678  | 12,000,000   | 97.2%        |\n",
    "| INPUT_rep2 | Rep2                | 23,456,789  | 23,000,000   | 98.1%        |\n",
    "| IP_rep1    | Rep1                | 34,567,890  | 34,000,000   | 98.4%        |\n",
    "| IP_rep2    | Rep2                | 45,678,901  | 45,000,000   | 98.6%        |"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "59f4cf06-9427-4485-a2a5-de669a1d3e90",
   "metadata": {},
   "source": [
    "To assess replicate concordance, we generated a sample correlation heatmap based on Pearson correlation of genome-wide coverage **(Figure 6)**. The clustering pattern was identical to the original Supplementary Figure S2B, with biological replicates clustering tightly together (INPUTs with INPUTs, IPs with IPs). The Pearson correlation was selected over Spearman because it measures the linear relationship between raw signal intensities, which is more appropriate for comparing coverage values from sequencing depth. However, our calculated correlation coefficients were lower than those reported in the paper (e.g., ~0.65 for IP replicates vs. 0.91 originally). This is likely due to our more stringent blacklist filtering and normalization methods, which remove artifactual signals that can inflate correlation, thereby providing a more conservative and biologically accurate estimate of replicate similarity."
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "7515ec6a-4e0a-442d-a7a4-1c830fb13228",
   "metadata": {},
   "source": [
    "![alt_text](Figure6_correlation_plot.png)\n",
    "**Figure 6. Matrix showing the Pearson correlation of the signal intensity between the IP and INPUT samples.**"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "3802cc7d-3db0-4e2a-9745-8f710cf66493",
   "metadata": {},
   "source": [
    "Further evaluating reproducibility, a Venn diagram was created to compare peak calls between biological replicates (Figure 7). analysis identified a larger number of total peaks in each replicate (Rep1: 11,103; Rep2: 17,958) compared to the original Figure S2C (Rep1: 3,983; Rep2: 10,465). This substantial difference can be explained: I used a more sensitive peak-caller algorithm (HOMER) with specific parameters tuned for a transcription factor, which is opposed to the potentially more conservative tool or parameters used in the original study. Additionally, the permissive definition of overlap (requiring only 1 bp) for defining reproducible peaks in my pipeline was designed to maximize sensitivity, further contributing to the larger peak set. Despite the quantitative differences, the key qualitative conclusion holds: a substantial subset of RUNX1 binding sites is reproducible across biological replicates, validating the overall success and reliability of the ChIP-seq experiment."
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "bd496fff-0a05-4481-ad62-efdaba8c3256",
   "metadata": {},
   "source": [
    "![alt_text](Figure7_venn_diagram.png)\n",
    "**Figure 7. Venn Diagram showing the RUNX1 peak reproducibility among the biological replicates.**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bd6f4946-ef87-4612-8a0b-5739c85e0b15",
   "metadata": {},
   "source": [
    "### Annotated Peaks Anlysis and Functional Networks of RUNX1 Target Genes\n",
    "Annotation of the 4,364 reproducible RUNX1 peaks revealed a strong enrichment for promoter-TSS regions, with nearly half (48.7%) of all peaks **(Table 2)**. This distribution is consistent with RUNX1's canonical role as a transcription factor directly regulating gene expression initiation. To elucidate the functional impact of this binding, I extracted the protein-coding genes associated with promoter-bound RUNX1 peaks and analyzed their network using the STRING database(13) **(Figure 8)**. The resulting protein-protein interaction network revealed significant clustering of genes involved in key cancer-related pathways, including regulators of the cell cycle, transcriptional regulation, and signal transduction. This analysis indicates that RUNX1 targets are not isolated entities but form coherent functional modules, suggesting a coordinated regulatory program where RUNX1 acts as a master regulator of oncogenic networks in MCF-7 cells."
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "e01e8bf7-dd4d-47c3-87c9-90817ef9ea20",
   "metadata": {},
   "source": [
    "| Annotation Type | Peak Number | Percentage (%) |\n",
    "|-----------------|-------------|----------------|\n",
    "| promoter-TSS    | 2123        | 48.66          |\n",
    "| intron          | 1200        | 27.5           |\n",
    "| Intergenic      | 727         | 16.66          |\n",
    "| TTS             | 158         | 3.62           |\n",
    "| exon            | 155         | 3.55           |"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "fb037e88-c53f-4dee-b0c3-fb4f327c1afe",
   "metadata": {},
   "source": [
    "![alt_text](Figure8_STRING.png)\n",
    "**STRING network analysis of protein–protein interactions among genes associated with promoter-bound RUNX1 peaks.**"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "ce36a4f4-5f74-4644-8687-a2745e83f28c",
   "metadata": {},
   "source": [
    "## Reference:\n",
    "(1)\tOkuda, T.; Nishimura, M.; Nakao, M.; Fujitaa, Y. RUNX1/AML1: A Central Player in Hematopoiesis. Int. J. Hematol. 2001, 74 (3), 252–257. https://doi.org/10.1007/BF02982057.\n",
    "(2)\tBarutcu, A. R.; Hong, D.; Lajoie, B. R.; McCord, R. P.; Van Wijnen, A. J.; Lian, J. B.; Stein, J. L.; Dekker, J.; Imbalzano, A. N.; Stein, G. S. RUNX1 Contributes to Higher-Order Chromatin Organization and Gene Regulation in Breast Cancer Cells. Biochim. Biophys. Acta BBA - Gene Regul. Mech. 2016, 1859 (11), 1389–1397. https://doi.org/10.1016/j.bbagrm.2016.08.003.\n",
    "(3)\tDi Tommaso, P.; Chatzou, M.; Floden, E. W.; Barja, P. P.; Palumbo, E.; Notredame, C. Nextflow Enables Reproducible Computational Workflows. Nat. Biotechnol. 2017, 35 (4), 316–319. https://doi.org/10.1038/nbt.3820.\n",
    "(4)\tKratzke, N. Lightweight Virtualization Cluster How to Overcome Cloud Vendor Lock-In. J. Comput. Commun. 2014, 02 (12), 1–7. https://doi.org/10.4236/jcc.2014.212001.\n",
    "(5)\tBolger, A. M.; Lohse, M.; Usadel, B. Trimmomatic: A Flexible Trimmer for Illumina Sequence Data. Bioinformatics 2014, 30 (15), 2114–2120. https://doi.org/10.1093/bioinformatics/btu170.\n",
    "(6)\tLangmead, B.; Salzberg, S. L. Fast Gapped-Read Alignment with Bowtie 2. Nat. Methods 2012, 9 (4), 357–359. https://doi.org/10.1038/nmeth.1923.\n",
    "(7)\tLi, H.; Handsaker, B.; Wysoker, A.; Fennell, T.; Ruan, J.; Homer, N.; Marth, G.; Abecasis, G.; Durbin, R.; 1000 Genome Project Data Processing Subgroup. The Sequence Alignment/Map Format and SAMtools. Bioinformatics 2009, 25 (16), 2078–2079. https://doi.org/10.1093/bioinformatics/btp352.\n",
    "(8)\tEwels, P.; Magnusson, M.; Lundin, S.; Käller, M. MultiQC: Summarize Analysis Results for Multiple Tools and Samples in a Single Report. Bioinformatics 2016, 32 (19), 3047–3048. https://doi.org/10.1093/bioinformatics/btw354.\n",
    "(9)\tRamírez, F.; Ryan, D. P.; Grüning, B.; Bhardwaj, V.; Kilpert, F.; Richter, A. S.; Heyne, S.; Dündar, F.; Manke, T. deepTools2: A next Generation Web Server for Deep-Sequencing Data Analysis. Nucleic Acids Res. 2016, 44 (W1), W160–W165. https://doi.org/10.1093/nar/gkw257.\n",
    "(10)\tHeinz, S.; Benner, C.; Spann, N.; Bertolino, E.; Lin, Y. C.; Laslo, P.; Cheng, J. X.; Murre, C.; Singh, H.; Glass, C. K. Simple Combinations of Lineage-Determining Transcription Factors Prime Cis-Regulatory Elements Required for Macrophage and B Cell Identities. Mol. Cell 2010, 38 (4), 576–589. https://doi.org/10.1016/j.molcel.2010.05.004.\n",
    "(11)\tQuinlan, A. R.; Hall, I. M. BEDTools: A Flexible Suite of Utilities for Comparing Genomic Features. Bioinformatics 2010, 26 (6), 841–842. https://doi.org/10.1093/bioinformatics/btq033.\n",
    "(12)\tThorvaldsdottir, H.; Robinson, J. T.; Mesirov, J. P. Integrative Genomics Viewer (IGV): High-Performance Genomics Data Visualization and Exploration. Brief. Bioinform. 2013, 14 (2), 178–192. https://doi.org/10.1093/bib/bbs017.\n",
    "(13)\tSzklarczyk, D.; Kirsch, R.; Koutrouli, M.; Nastou, K.; Mehryary, F.; Hachilif, R.; Gable, A. L.; Fang, T.; Doncheva, N. T.; Pyysalo, S.; Bork, P.; Jensen, L. J.; von Mering, C. The STRING Database in 2023: Protein–Protein Association Networks and Functional Enrichment Analyses for Any Sequenced Genome of Interest. Nucleic Acids Res. 2023, 51 (D1), D638–D646. https://doi.org/10.1093/nar/gkac1000."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "516215d2-97b4-4d2d-ab16-b0d486aa7da4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:base] *",
   "language": "python",
   "name": "conda-base-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
